ports/nrf/modules/machine/spi.c

```diff
@@ -248,75 +227,85 @@ STATIC MP_DEFINE_CONST_DICT(machine_spi_locals_dict, machine_spi_locals_dict_tab
-        self->spi->init.mosi_pin = &MICROPY_HW_SPI0_MOSI;
-        self->spi->init.miso_pin = &MICROPY_HW_SPI0_MISO;
+        config.sck_pin  = (&MICROPY_HW_SPI0_SCK)->pin;
+        config.mosi_pin = (&MICROPY_HW_SPI0_MOSI)->pin;
+        config.miso_pin = (&MICROPY_HW_SPI0_MISO)->pin;
     }
 
+    // Manually trigger slave select from upper layer.
+    config.ss_pin = NRFX_SPI_PIN_NOT_USED;
+
     int baudrate = args[ARG_NEW_baudrate].u_int;
 
     if (baudrate <= 125000) {
-        self->spi->init.freq = HAL_SPI_FREQ_125_Kbps;
+        config.frequency = NRF_SPI_FREQ_125K;
     } else if (baudrate <= 250000) {
-        self->spi->init.freq = HAL_SPI_FREQ_250_Kbps;
+        config.frequency = NRF_SPI_FREQ_250K;
     } else if (baudrate <= 500000) {
-        self->spi->init.freq = HAL_SPI_FREQ_500_Kbps;
+        config.frequency = NRF_SPI_FREQ_500K;
     } else if (baudrate <= 1000000) {
-        self->spi->init.freq = HAL_SPI_FREQ_1_Mbps;
+        config.frequency = NRF_SPI_FREQ_1M;
     } else if (baudrate <= 2000000) {
-        self->spi->init.freq = HAL_SPI_FREQ_2_Mbps;
+        config.frequency = NRF_SPI_FREQ_2M;
     } else if (baudrate <= 4000000) {
-        self->spi->init.freq = HAL_SPI_FREQ_4_Mbps;
+        config.frequency = NRF_SPI_FREQ_4M;
     } else if (baudrate <= 8000000) {
-        self->spi->init.freq = HAL_SPI_FREQ_8_Mbps;
+        config.frequency = NRF_SPI_FREQ_8M;
 #if NRF52840_XXAA
     } else if (baudrate <= 16000000) {
-        self->spi->init.freq = HAL_SPI_FREQ_16_Mbps;
+        config.frequency = SPIM_FREQUENCY_FREQUENCY_M16; // Temporary value until SPIM support is addressed (EasyDMA)
     } else if (baudrate <= 32000000) {
-        self->spi->init.freq = HAL_SPI_FREQ_32_Mbps;
+        config.frequency = SPIM_FREQUENCY_FREQUENCY_M32; // Temporary value until SPIM support is addressed (EasyDMA)
 #endif
     } else { // Default
-        self->spi->init.freq = HAL_SPI_FREQ_1_Mbps;
+        config.frequency = NRF_SPI_FREQ_1M;
     }
+
 #ifdef NRF51
-    self->spi->init.irq_priority = 3;
+    config.irq_priority = 3;
 #else
-    self->spi->init.irq_priority = 6;
+    config.irq_priority = 6;
 #endif
-    self->spi->init.mode = HAL_SPI_MODE_CPOL0_CPHA0;
-    self->spi->init.firstbit = (args[ARG_NEW_firstbit].u_int == 0) ? HAL_SPI_MSB_FIRST : HAL_SPI_LSB_FIRST;;
-    hal_spi_master_init(self->spi->instance, &self->spi->init);
+
+    config.mode = NRF_SPI_MODE_0;
+    config.orc  = 0xFF; // Overrun character
+    config.bit_order = (args[ARG_NEW_firstbit].u_int == 0) ? NRF_SPI_BIT_ORDER_MSB_FIRST : NRF_SPI_BIT_ORDER_LSB_FIRST;
+
+    // Set context to this instance of SPI
+    nrfx_spi_init(&self->p_spi, &config, NULL, (void *)self);
 
     return MP_OBJ_FROM_PTR(self);
 }
 
 STATIC void machine_hard_spi_init(mp_obj_t self_in, mp_arg_val_t *args) {
+    // TODO
 }
 
 STATIC void machine_hard_spi_deinit(mp_obj_t self_in) {
-    machine_hard_spi_obj_t *self = self_in;
-    spi_deinit(self->spi);
+    // TODO
```

Comment:
If `SPI.init` and `SPI.deinit` that call the above functions (`machine_hard_spi_init` and `machine_hard_spi_deinit`) don't actually do anything, why provide them? Maybe it's less confusing if they are (temporarily) left out.